helix-term/src/commands.rs

```diff
@@ -563,6 +567,16 @@ fn extend_to_line_start(cx: &mut Context) {
     goto_line_start_impl(view, doc, Movement::Extend)
 }
 
+fn kill_to_line_start(cx: &mut Context) {
+    extend_to_line_start(cx);
+    delete_selection(cx);
```

Comment:
You probably want to use `delete_selection_impl` for these: https://github.com/helix-editor/helix/blob/9e458272b16a7208e398158bd07a7caa08801597/helix-term/src/commands.rs#L1547-L1573

`delete_selection` will also store all the changes into history, but in insert mode we don't want to do that until insert mode is exited.

Reply:
Is it OK if I add a flag in delete_selection as it contains some common logic? Or it's better that I wrote a wrapper like current delete_selection?
```rust
fn delete_selection(cx: &mut Context) {
    delete_selection_and_append_history(cx, true);
}

fn delete_selection_and_append_history(cx: &mut Context, append_change_to_history: bool) {
    let reg_name = cx.register.unwrap_or('"');
    let (view, doc) = current!(cx.editor);
    let registers = &mut cx.editor.registers;
    let reg = registers.get_mut(reg_name);
    delete_selection_impl(reg, doc, view.id);

    if append_change_to_history {
        doc.append_changes_to_history(view.id);
    }

    // exit select mode, if currently in select mode
    exit_select_mode(cx);
}
```

Reply:
So using only `delete_selection_impl` should be enough

Reply:
Done, thanks for the finding, please take another look.